lighthouse-core/config/config.js

```diff
@@ -162,6 +161,34 @@ function assertValidGatherer(gathererInstance, gathererName) {
   }
 }
 
+
+/**
+ * Validate the LH.Flags
+ * @param {LH.Flags} flags
+ */
+function assertValidFlags(flags) {
+  // @ts-expect-error Checking for removed flags
+  if (flags.emulatedFormFactor || flags.internalDisableDeviceScreenEmulation) {
+    throw new Error('Invalid emulation flag. Emulation configuration changed in LH 7.0. See https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md'); // eslint-disable-line max-len
+  }
+}
+
+/**
+ * Validate the settings after they've been built
+ * @param {LH.Config.Settings} settings
+ */
+function assertValidSettings(settings) {
+  if (!settings.formFactor) {
+    throw new Error(`\`settings.formFactor\` must be defined as 'mobile' or 'desktop'. See https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md`); // eslint-disable-line max-len
+  }
+
+  if (settings.screenEmulation) {
+    if (settings.screenEmulation.mobile !== (settings.formFactor === 'mobile')) {
+      log.error('config', `Screen emulation mobile setting (${settings.screenEmulation.mobile}) does not match formFactor setting (${settings.formFactor})`); // eslint-disable-line max-len
```

Comment:
what are the cases we want to support by not doing a fatal error here?

Reply:
hmmmmmmm i'm fine with going fatal. i can't think of a reasonable usecase.

Reply:
Fatal errors seem better than weird unexpected results. LGTM